controllers/apidefinition_controller.go

```diff
@@ -285,6 +288,47 @@ func (r *ApiDefinitionReconciler) updateLinkedPolicies(ctx context.Context, a *t
 	}
 	return
 }
+func (r *ApiDefinitionReconciler) updateLoopingTargets(ctx context.Context, a *tykv1alpha1.ApiDefinition) {
+	if a.Spec.Proxy.TargeInternal != nil {
+		a.Spec.Proxy.TargetURL = formatLoop(a.Spec.Proxy.TargeInternal)
+		a.Spec.Proxy.TargeInternal = nil
+	}
+	d := &a.Spec.VersionData
+	for n := range d.Versions {
+		v := d.Versions[n]
+		if v.ExtendedPaths != nil {
+			for i := 0; i < len(v.ExtendedPaths.URLRewrite); i++ {
+				u := &v.ExtendedPaths.URLRewrite[i]
+				if u.RewriteToInternal != nil {
+					u.RewriteTo = formatLoop(u.RewriteToInternal)
+					u.RewriteToInternal = nil
+				}
+				for j := 0; j < len(u.Triggers); j++ {
+					x := &u.Triggers[j]
+					if x.RewriteToInternal != nil {
+						x.RewriteTo = formatLoop(x.RewriteToInternal)
+						x.RewriteToInternal = nil
+					}
+				}
+			}
+		}
+		d.Versions[n] = v
+	}
+}
```

Comment:
Would be great to see some tests - specifically for this loop.

Reply:
done f6c706b